pages/dao/[symbol]/proposal/components/instructions/CreateAssociatedTokenAccount.tsx

```diff
@@ -0,0 +1,174 @@
+    GovernedMultiTypeAccount[]
+  >([])
+  const { getGovernedMultiTypeAccounts } = useGovernanceAssets()
+
+  useEffect(() => {
+    getGovernedMultiTypeAccounts().then(setGovernedAccounts)
+  }, [])
+
+  const shouldBeGoverned = index !== 0 && governance
+  const programId: PublicKey | undefined = realmInfo?.programId
+  const [form, setForm] = useState<CreateAssociatedTokenAccountForm>({})
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
+    setFormErrors(validationErrors)
+    return isValid
+  }
+
+  async function getInstruction(): Promise<UiInstruction> {
+    const isValid = await validateInstruction()
+
+    if (
+      !connection ||
+      !isValid ||
+      !programId ||
+      !form.governedAccount?.governance?.account ||
+      !form.splTokenMintUIName ||
+      !wallet?.publicKey
+    ) {
+      return {
+        serializedInstruction: '',
+        isValid: false,
+        governance: form.governedAccount?.governance,
+      }
+    }
+
+    const [tx] = await createAssociatedTokenAccount(
+      // fundingAddress
+      wallet.publicKey,
+
+      // walletAddress
+      form.governedAccount.governance.pubkey,
+
+      // splTokenMintAddress
+      getSplTokenMintAddressByUIName(form.splTokenMintUIName)
+    )
+
+    console.log('infos', {
+      governanceAccount: form.governedAccount.governance.account,
+      governanceOwner: form.governedAccount.governance.owner.toString(),
+      fundingAddress: wallet.publicKey.toString(),
+      walletAddress: form.governedAccount.governance.pubkey.toString(),
+      splTokenMintName: form.splTokenMintUIName,
+      splTokenMintAddress: getSplTokenMintAddressByUIName(
+        form.splTokenMintUIName
+      ).toString(),
+    })
+
+    console.log('tx', tx)
+
+    return {
+      serializedInstruction: serializeInstructionToBase64(tx),
+      isValid: true,
+      governance: form.governedAccount.governance,
+    }
+  }
+
+  useEffect(() => {
+    handleSetForm({
+      propertyName: 'programId',
+      value: programId?.toString(),
+    })
+  }, [realmInfo?.programId])
+
+  useEffect(() => {
+    handleSetInstructions(
+      {
+        governedAccount: form.governedAccount?.governance,
+        getInstruction,
+      },
+      index
+    )
+  }, [form])
+
+  const schema = yup.object().shape({
+    governedAccount: yup
+      .object()
+      .nullable()
+      .required('Governed account is required'),
+    splTokenMintUIName: yup.string().required('SPL Token Mint is required'),
+  })
+
+  return (
+    <>
+      <GovernedAccountSelect
+        label="Governance"
+        governedAccounts={governedAccounts as GovernedMultiTypeAccount[]}
```

Comment:
can cast closer to the data structure rather than where it's used ?

Reply:
with the latest fix on governance select, this cast doesn't exist anymore